### ISSUE TITLE
Wheels lagging behind car body using LiDAR (CARLA_0.9.8)

### ISSUE THREAD

**Issue by a-boinet** [carla-simulator/carla#2716]
Wheels seem to move with _at least_ a one frame delay after the car body, as shown below :

![image(2)](https://user-images.githubusercontent.com/30056175/78982705-fe133e80-7b22-11ea-9d4d-e756d87a72e6.png)

During acceleration, _the shift is even more important_ : 

![tmp(1)](https://user-images.githubusercontent.com/30056175/78982166-d4a5e300-7b21-11ea-96b4-6a90d1d5dc9e.png)

When the car first starts, its wheels are at the back of the body.
If the car accelerates, the wheels start lagging behind the body.
And when the car stops, the wheels eventually catch up after a few frames.

I'm running on carla 0.9.8, in synchronous mode at 10 fps.
LiDAR attributes are : 
```python
lidar_attributes = {
        "channels": 128,
        "range": 75,  # In carla 0.9.7 This is meters !!!!
        "points_per_second": 4800000,  # 100000
        "rotation_frequency": 20,
        "upper_fov": 15.0,
        "lower_fov": -25.0,
        "sensor_tick": 0.0,
    }
# The position of the LiDAR sensor in the host vehicle reference frame is
x=0, y=0, z=2.5
```

My guess is that the wheels collision boxes and the body collision box aren't updated at the same time, but that's anyone's guess

**Comment by marcgpuig** [comment]
Hi @a-boinet 
Thanks for your feedback, we will look into that, this could be caused because of how UE4 handles the ray cast. Maybe using different flags for ray casting could solve it. It is UE 4.22 right?

**Comment by a-boinet** [comment]
Hi @marcgpuig 
Thanks for your reply. It is UE 4.22 indeed.
I'll test this on CARLA 0.9.9 asap to see if it makes any difference.

**Comment by a-boinet** [comment]
It looks _exactly_ the same on CARLA 0.9.9 (UE 4.24) :

![image(3)](https://user-images.githubusercontent.com/30056175/80722986-4c9f7180-8b00-11ea-93fc-88bb359ff2ec.png)

**Comment by marcgpuig** [comment]
@a-boinet Thanks for your quick testing.

This has been also commented in #1670 and may be slightly related to #423.

We need to investigate why the wheels are behind the car collision box, whether it's how the physics engine tick works or if we can really do something about it. This search could also be extended to find a suitable ray cast emission that can correctly detect UE4's `SkeletalMeshe` and not just the car's bounding box, so it can detect the shape of the car correctly.
We definitely need to find some time to address it.

This could also be happening with the radar sensor as it uses the same physics parameters, but I have not tested it yet.

**Comment by marcgpuig** [comment]
Hi @a-boinet sorry for the late update.
We're improveing the vehicle collision meshes and now look way better, with these changes this wheel problem should be fixed too!

![image](https://user-images.githubusercontent.com/15848712/85411390-9472d000-b568-11ea-9e4e-130e6df4dec6.png)
![image](https://user-images.githubusercontent.com/15848712/85411831-27ac0580-b569-11ea-9337-c7b2bc10c04e.png)

**Comment by a-boinet** [comment]
Looks great! Thanks for the update, I'm looking forward to testing this

**Comment by UILXELA** [comment]
> Hi @a-boinet sorry for the late update.
> We're improving the vehicle collision meshes and now looks way better, with these changes this wheel problem should be fixed too!
> 
> ![image](https://user-images.githubusercontent.com/15848712/85411390-9472d000-b568-11ea-9e4e-130e6df4dec6.png)
> ![image](https://user-images.githubusercontent.com/15848712/85411831-27ac0580-b569-11ea-9337-c7b2bc10c04e.png)

I had this issue too. If in the new version lidar-captured cars look like real cars instead of boxes, that would be a big progress. Thank you for the update.

**Comment by pergustav** [comment]
Hi,

I face the same problem as stated in this issue, with Carla 0.9.10. I have custom vehicle, and the wheels lag behind the body, when driving. Followed [this](https://carla.readthedocs.io/en/latest/tuto_D_generate_colliders/#physics-colliders) tutorial to create the mesh.

I have checked the official Cyber Truck, and the wheels are static to the body. The Low density mesh is one piece, I assume. I have made a small record for demonstrating. 

I think the problem actually still exists. 

https://user-images.githubusercontent.com/49983400/112976987-d2370180-9155-11eb-8b95-48fb68820b1f.mp4
